hazelcast-sql/src/test/java/com/hazelcast/jet/index/JetSqlIndexAbstractTest.java

```diff
@@ -0,0 +1,709 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+import static com.hazelcast.jet.sql.impl.connector.map.OnHeapMapIndexScanP.OnHeapMapIndexScanMetaSupplier;
+import static com.hazelcast.sql.impl.SqlTestSupport.getLocalKeys;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.and;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.eq;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.eq_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.gt;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.gt_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.gte;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.gte_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.isNotNull;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.isNotNull_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.isNull;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.isNull_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.lt;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.lt_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.lte;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.lte_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.neq;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.neq_2;
+import static com.hazelcast.sql.support.expressions.ExpressionPredicates.or;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.BIG_DECIMAL;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.BIG_INTEGER;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.BOOLEAN;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.BYTE;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.CHARACTER;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.DOUBLE;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.FLOAT;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.INTEGER;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.LONG;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.SHORT;
+import static com.hazelcast.sql.support.expressions.ExpressionTypes.STRING;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings({"unchecked", "rawtypes", "unused"})
+public abstract class JetSqlIndexAbstractTest extends SqlTestSupport {
```

Comment:
This class tests specifically `IMap` behaviour. I would move it to `IMap` specific package.
Moreover, let's make sure Jet engine is used by this test by using a Jet engine feature that is not supported by IMDG engine.